### PR TITLE
Add some further bash-completion

### PR DESCRIPTION
**Pull request by cwayne18**
Autocomplete the plugins after 'help'
Autocomplete directory names after 'snap' or 'tour'
Autocomplete all '.snap' files after 'gated, upload, push, sign-build, register' or 'validate'

### CODE REVIEW

**Comment by elopio** [comment]
Hey @cwayne18, I have a question and since you are our autocomplete expert... How would autocomplete work if snapcraft is not installed as a deb? For example, if it comes from a snap, or pip.

**Comment by cwayne18** [comment]
@elopio I suppose for pip we could include the bash-completion file in data_files in setup.py, as for snaps, not entirely sure the best way to go about it yet

**Comment by sergiusens** [comment]
Sorry for taking so long to get to this. `data_files` is a rather dead concept in the `pip` world, if you use it, it would land in `site-packages` going forward.

For snapcraft as a snap we need to have some form of feature work with the snappy core team.

File: debian/snapcraft-completion [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
+    help)
+        plugins=$(snapcraft list-plugins)
+        COMPREPLY=( $(compgen -W "$plugins" -- $cur))
+        return 0
+        ;;
+    snap | tour)
+        _filedir -d
+        return 0
+        ;;
+    gated | upload | push | sign-build | register | validate)
```
Comment (sergiusens) [inline debian/snapcraft-completion]: gated register and validate shouldn't use snap-file names.

**Pull request merged**